Userland/Utilities/headless-browser.cpp

```diff
@@ -0,0 +1,696 @@
+                    weak_this->on_error(Web::WebSockets::WebSocketClientSocket::Error::ConnectionUpgradeFailed);
+                    return;
+                case WebSocket::WebSocket::Error::ServerClosedSocket:
+                    weak_this->on_error(Web::WebSockets::WebSocketClientSocket::Error::ServerClosedSocket);
+                    return;
+                }
+                weak_this->on_error(Web::WebSockets::WebSocketClientSocket::Error::CouldNotEstablishConnection);
+            };
+            m_websocket->on_close = [weak_this = make_weak_ptr()](u16 code, String reason, bool was_clean) {
+                weak_this->on_close(code, move(reason), was_clean);
+            };
+        }
+
+        NonnullRefPtr<WebSocket::WebSocket> m_websocket;
+    };
+
+    static NonnullRefPtr<HeadlessWebSocketClientManager> create()
+    {
+        return adopt_ref(*new HeadlessWebSocketClientManager());
+    }
+
+    virtual ~HeadlessWebSocketClientManager() override { }
+
+    virtual RefPtr<Web::WebSockets::WebSocketClientSocket> connect(AK::URL const& url, String const& origin) override
+    {
+        WebSocket::ConnectionInfo connection_info(url);
+        connection_info.set_origin(origin);
+
+        auto connection = HeadlessWebSocket::create(WebSocket::WebSocket::create(move(connection_info)));
+        return connection;
+    }
+
+private:
+    HeadlessWebSocketClientManager() { }
+};
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    int take_screenshot_after = 1;
+    StringView url;
+    StringView fonts_database;
+    StringView favicon_path;
+    StringView error_page_url;
+
+    Core::EventLoop event_loop;
+    Core::ArgsParser args_parser;
+    args_parser.set_general_help("This utility runs the Browser in headless mode.");
+    args_parser.add_option(take_screenshot_after, "Take a screenshot after [n] seconds (default: 1)", "screenshot", 's', "n");
+    args_parser.add_option(fonts_database, "Path of the fonts on your system", "fonts", 'f', "font-database-path");
+    args_parser.add_option(favicon_path, "Path of the default favicon", "favicon", 'i', "default-favicon-path");
+    args_parser.add_option(error_page_url, "URL for the error page", "error-page", 'e', "error-page-url");
+    args_parser.add_positional_argument(url, "URL to open", "url", Core::ArgsParser::Required::Yes);
+    args_parser.parse(arguments);
+
+    Web::ImageDecoding::Decoder::initialize(HeadlessImageDecoderClient::create());
+    Web::ResourceLoader::initialize(HeadlessRequestServer::create());
+    Web::WebSockets::WebSocketClientManager::initialize(HeadlessWebSocketClientManager::create());
+
+    if (!favicon_path.is_empty()) {
+        Web::FrameLoader::set_default_favicon_path(favicon_path);
+    }
+
+    if (!fonts_database.is_empty()) {
+        Gfx::FontDatabase::set_default_fonts_lookup_path(fonts_database);
+        Gfx::FontDatabase::set_default_font_query("Katica 10 400 0");
```

Comment:
You should also set the monospace font here
```suggestion
        Gfx::FontDatabase::set_default_font_query("Katica 10 400 0");
        Gfx::FontDatabase::set_fixed_width_font_query("Csilla 10 400 0");
```